src/main/java/gg/solarmc/kitpvp/kill/DamageListener.java

```diff
@@ -0,0 +1,90 @@
+package gg.solarmc.kitpvp.kill;
+
+import gg.solarmc.kitpvp.KitpvpConfig;
+import gg.solarmc.kitpvp.kill.damage.DamageMap;
+import gg.solarmc.kitpvp.messaging.MessageConfig;
+import gg.solarmc.kitpvp.messaging.MessageController;
+import gg.solarmc.kitpvp.messaging.parsers.PairPlayerParser;
+import gg.solarmc.kitpvp.messaging.parsers.SinglePlayerParser;
+import gg.solarmc.loader.DataCenter;
+import gg.solarmc.loader.kitpvp.KitPvpKey;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import space.arim.omnibus.util.concurrent.CentralisedFuture;
+
+import java.util.Optional;
+import java.util.Set;
+
+public class DamageListener implements Listener {
+
+    private final DamageMap damageMap;
+    private final KillDataHandler killDataHandler;
+    private final MessageConfig messageConfig;
+    private final KitpvpConfig config;
+
+    public DamageListener(DamageMap damageMap, KillDataHandler killDataHandler, MessageConfig messageConfig, KitpvpConfig config) {
+        this.damageMap = damageMap;
+        this.killDataHandler = killDataHandler;
+        this.messageConfig = messageConfig;
+        this.config = config;
+    }
+
+    @EventHandler //honk
+    public void onDamage(EntityDamageByEntityEvent event) {
+        if (event.getEntity() instanceof Player && event.getDamager() instanceof Player) {
```

Comment:
```suggestion
        if (event.getEntity() instanceof Player damaged && event.getDamager() instanceof Player damager) {
```
instanceof pattern matching is a fun feature which will allow you to get rid of the next 2 lines

Reply:
Since when was that a feature?

Reply:
Preview feature in JDK 15, full feature in JDK 16